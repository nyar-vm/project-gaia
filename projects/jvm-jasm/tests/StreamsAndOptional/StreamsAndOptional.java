import java.util.*;
import java.util.stream.*;

public class StreamsAndOptional {
    
    public Optional<String> findFirst(List<String> list) {
        return list.stream()
                  .filter(s -> s.length() > 3)
                  .findFirst();
    }
    
    public IntStream primitiveStream() {
        return IntStream.of(1, 2, 3, 4, 5)
                       .map(x -> x * 2)
                       .filter(x -> x > 4);
    }
    
    public Stream<String> flatMapExample() {
        List<List<String>> nested = Arrays.asList(
            Arrays.asList("a", "b"),
            Arrays.asList("c", "d")
        );
        
        return nested.stream()
                    .flatMap(List::stream)
                    .map(String::toUpperCase);
    }
    
    public OptionalInt maxOptional() {
        return IntStream.empty().max();
    }
    
    public void streamChaining() {
        List<String> result = Arrays.stream(new String[]{"hello", "world", "java"})
            .peek(s -> System.out.println("Processing: " + s))
            .filter(s -> s.length() > 4)
            .map(String::toUpperCase)
            .collect(Collectors.toList());
    }
}