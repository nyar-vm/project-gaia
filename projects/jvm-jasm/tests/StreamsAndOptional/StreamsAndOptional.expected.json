{
    "class_name": "StreamsAndOptional",
    "class_modifiers": [
        "public",
        "super"
    ],
    "method_count": 2,
    "field_count": 0,
    "expected_methods": [
        "<init>:()V",
        "findFirst:(Ljava/util/List;)Ljava/util/Optional;"
    ],
    "expected_fields": [],
    "syntactic_sugar_features": [],
    "check_tokens": true,
    "source_file": null
}